### ISSUE TITLE
Panda fails to update from v2.1 to v3.0 using Web GUI interface

### ISSUE THREAD

**Issue by dmgav** [PandABlocks/PandABlocks-rootfs#43]
Panda fails to update rootfs package from v2.1 to v3.0. Used package:

[boot@PandABox-3.0.zip](https://github.com/PandABlocks/PandABlocks.github.io/releases/download/3.0/boot@PandABox-3.0.zip) (downloaded from https://github.com/PandABlocks/PandABlocks.github.io/releases)

After the update was started from Web GUI, the updated failed with the following message:
```
Unzipping /mnt/sda1/boot@PandABox-3.0.zip...

Archive: /mnt/sda1/boot@PandABox-3.0.zip
    creating: boot/
unzip: zip flags 1 and 8 are not supported
```

This issue was observed while updating 4 Panda units (all units that we currently have). I was able to successfully update all 4 Pandas by unpacking the contents of the package directly to the boot partition and restarting Panda. Additionally, on 2 Pandas I was not able to install any packages using Web GUI following 'rootfs' update and attempt to update FPGA packages first, and had to format the SD card and install all packages from scratch. There are no issues with the software, since all Panda units work correctly after the update. It may be an issue with packages, which are incompatible with the server v2.1.

**Comment by dmgav** [comment]
@tomtrafford I confirm that the new version of `boot@PandABox-3.0.zip` can be properly install using the web GUI, so this particular issue is resoved. Nevertheless the rest of the upgrade using web GUI did not go smoothly and I had to resort to formatting SD card and installing all packages from scratch.

This is the sequence of steps I did:

- I did not uninstall any packages.
- The new packages (.zpg) were copied on the flash drive.
- I selected all new packages in the web GUI and clicked 'Install Packages'.
- The first of the selected packages ('server') was installed successfully, but then installation stopped.
- I tried to install other packages one by one, but installation failed for all packages. After attempt to install FPGA package Panda became unresponsive and remained unresponsive after power cycling. 

We have a few more Pandas that we plan to upgrade, so there is opportunity to do some testing that may help other users. Is there any specific sequence of steps I could try? Should I remove old packages first? Or is there specific order in which packages should be upgraded.

**Comment by tomtrafford** [comment]
I will have a go at trying to replicate this. Did you see any errors when the package installations failed?

**Comment by dmgav** [comment]
Yes, there were errors, but unfortunately I did not document them. I was hoping that brute force trial-and-error method would get me through this time, but it didn't.

**Comment by tomtrafford** [comment]
It looks like this problem occurs only if the panda-server is updated before the panda-fpga. If you either remove or update the exiting panda-fpga package before installing the other packages, you should not encounter this problem. This is not expected functionality, the PandA server should check for a Flag in the FPGA Config to ensure it is the correct version, but this does seem to be functioning as expected.

On a PandA where you have encountered this problem, you can use a Config overide file on the USB stick, panda-config.txt as described here: https://pandablocks.github.io/PandABlocks-rootfs/master/quickstart.html

If you uncomment the line NO_ZPKG, the PandA won't attempt to load any zpkgs on startup and you can recover from this behaviour by installing the FPGA zpkg. (And then re-comment the NO_ZPKG, and restart the PandA).

**Comment by dmgav** [comment]
Thank you. I will try updating the FPGA package first with the next Panda and let you know if it went smoothly. 
I think I tried commenting out `NO_ZPKG` line with one of the Pandas and it would let me to connect with SSH, but I was unable to install any packages. I don't remember if I specifically tried installing the FPGA package.

**Comment by dmgav** [comment]
I confirm that Panda can be updated from v2.1 to v3.0 using web GUI. I guess the key is to install the FPGA package first. I used the following sequence:
- FPGA package;
- slow-FPGA;
- server;
- remaining web-server/client packages.
